lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -476,16 +482,11 @@ private StateIndication checkSuspend(StateIndication stateChange) {
 
 		/* FIXME: we might want to limit this behaviour to only a specific
 		 * set of error codes */
-		if(pendingConnect != null && !pendingConnect.fallback && checkConnectivity()) {
-			String[] fallbackHosts = Defaults.getFallbackHosts(options);
-			if(fallbackHosts != null && fallbackHosts.length > 0) {
-				/* we will try a fallback host */
-				StateIndication fallbackConnectRequest = new StateIndication(ConnectionState.connecting, null);
-				fallbackConnectRequest.useFallbackHost = true;
-				requestState(fallbackConnectRequest);
-				/* returning null ensures we stay in the connecting state */
-				return null;
-			}
+		if(pendingConnect != null && checkConnectivity()) {
```

Comment:
Perhaps this is outside of this PR, but firstly you are not checking if fallback hosts are supported here, but secondly you are performing an internet up check for every single retry which is unnecessary.  If the primary host fails, you check that the internet is up, and if it is, you try all the fallback hosts up to the max number.  There is no need to keep checking if the internet is up between each fallback host attempt.